### PR TITLE
Prevent Transferring to 0x0 and Contract Address

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
Prevent transferring to the 0x0 address, and the contract address, according to Token Implementation Best Practices.
Reference: https://consensys.github.io/smart-contract-best-practices/tokens/

### CODE REVIEW

**Comment by maurelian** [comment]
@Muhammad-Altabba I'd ask you to also implement tests to verify the behaviour you've introduced, but first...

@skmgoldin If we added these protections (ie. prevent send to `0x0` and `address(this)`), I think we should do so in a child contract that inherits from EIP20.sol. What do you think?

**Comment by Muhammad-Altabba** [comment]
I will be happy to implement tests to verify the behavior. And if we need to add the proposed check in a child contract I will also be happy to implement (even I prefer it to be inside the same EIP20.sol contract).
Please direct me with your recommendation and I will do the needed.
Thanks,

**Comment by skmgoldin** [comment]
I feel it definitely should not be in EIP20.sol. Perhaps a different implementation called something like SafeEIP20.sol or something.

**Comment by Muhammad-Altabba** [comment]
Dear @maurelian and @skmgoldin,
I added SafeEIP20.sol and changed EIP20Factory to use SafeEIP20. And I wrote JavaScript Truffle Tests for the new contract: SafeEIP20.
Please let me know if you think that more things are needed to be done or modified.
Thanks,

File: test/eip20/safeEip20.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  it('transfers-from: should fail when the recipient is the contract address.', async () => {
+    await HST.approve(accounts[1], 100, { from: accounts[0] });
+    const allowance = await HST.allowance.call(accounts[0], accounts[1]);
+    assert.strictEqual(allowance.toNumber(), 100);
+
+    await assertRevert(HST.transferFrom.call(accounts[0], HST.address, 10, { from: accounts[1] }));
+  });   
+  
+  
+  //Starting from this line, the following tests are the same tests applied to EIP20.
```
Comment (maurelian) [inline test/eip20/safeEip20.js]: Is there a way that these tests could be exported from the other test file, and run on the safeEIP20 implementation? I'd prefer not to have to copy/paste code changes made from one to the next.

Comment (Muhammad-Altabba) [reply test/eip20/safeEip20.js]: Upon your comment, I Encapsulated the EIP compatible functions' tests inside a function and make it called for both EIP20 and SafeEIP20 smart contracts. And the duplicated tests code at safeEip20.js has been deleted.
So now, there is no copy/past code.

**Comment by Muhammad-Altabba** [comment]
Hello,
Is there anything, I can help with, to speed up accepting this Pull Request?
Thanks,

**Comment by Muhammad-Altabba** [comment]
Hi,
Could I help regarding this pull request?

**Comment by maurelian** [comment]
Done. Apologies for the delay Muhammad.

**Comment by Muhammad-Altabba** [comment]
No problem,
Many thanks,

**Pull request merged**